lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/CommunicationGroupClient.cs

```diff
@@ -175,5 +177,18 @@ public IScatterReceiver<T> GetScatterReceiver<T>(string operatorName)
 
             return (T) op;
         }
+
+        /// <summary>
+        /// Call each Operator to easure all the nodes in the topology group has been registered
+        /// </summary>
+        void ICommunicationGroupClientInternal.WaitingForRegistration()
+        {
+            foreach (var op in _operators.Values)
+            {
+                var methods = op.GetType().GetMethods(BindingFlags.NonPublic | BindingFlags.Instance);
+                var queriedMethods = methods.Where(e => e.Name.Equals("Org.Apache.REEF.Network.Group.Operators.IGroupCommOperatorInternal.WaitForRegistration")).ToArray();
+                queriedMethods[0].Invoke(op, null);
```

Comment:
IGroupCommOperatorInternal cannot extend IGroupCommOperator because we have one more layer like IBroadcastReceiver, IBroadcastSender which are public. They cannot extend from internal interface. 
The reason I am using reflection is because we have no knowledge of T for the Operators in operator dictionary. They are objects. Even if we have a way to use a proper interface, because we don't know T, we are still not able to call the method.